azurerm/internal/services/hpccache/hpc_cache_blob_nfs_target_resource.go

```diff
@@ -0,0 +1,242 @@
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.StorageTargetName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"cache_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"namespace_path": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.CacheNamespacePath,
+			},
+
+			"storage_container_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: storageValidate.StorageContainerResourceManagerID,
+			},
+
+			// TODO: use SDK enums once following issue is addressed
+			// https://github.com/Azure/azure-rest-api-specs/issues/13839
+			"usage_model": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"READ_HEAVY_INFREQ",
+					"READ_HEAVY_CHECK_180",
+					"WRITE_WORKLOAD_15",
+					"WRITE_AROUND",
+					"WRITE_WORKLOAD_CHECK_30",
+					"WRITE_WORKLOAD_CHECK_60",
+					"WRITE_WORKLOAD_CLOUDWS",
+				}, false),
+			},
+
+			"access_policy_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "default",
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+}
+
+func resourceHPCCacheBlobNFSTargetCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).HPCCache.StorageTargetsClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	cache := d.Get("cache_name").(string)
+	id := parse.NewStorageTargetID(subscriptionId, resourceGroup, cache, name)
+
+	if d.IsNewResource() {
+		resp, err := client.Get(ctx, id.ResourceGroup, id.CacheName, id.Name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return tf.ImportAsExistsError("azurerm_hpc_cache_blob_nfs_target", id.ID())
+		}
+	}
+
+	namespacePath := d.Get("namespace_path").(string)
+	containerId := d.Get("storage_container_id").(string)
+
+	// Construct parameters
+	namespaceJunction := []storagecache.NamespaceJunction{
+		{
+			NamespacePath:   &namespacePath,
+			TargetPath:      utils.String("/"),
+			NfsExport:       utils.String("/"),
+			NfsAccessPolicy: utils.String(d.Get("access_policy_name").(string)),
+		},
+	}
+	param := &storagecache.StorageTarget{
+		StorageTargetProperties: &storagecache.StorageTargetProperties{
+			Junctions:  &namespaceJunction,
```

Comment:
should we just inline this instead of creating a var for it?

Reply:
The `namespaceJunction` is a slice, for which we do not have a utility function to get its pointer. That's the reason to declare a local var here.